libsolidity/ast/ASTUtils.cpp

```diff
@@ -19,12 +19,35 @@
 #include <libsolidity/ast/AST.h>
 #include <libsolidity/ast/ASTUtils.h>
 
+#include <libsolutil/Algorithms.h>
+
 namespace solidity::frontend
 {
 
+bool isConstantVariableRecursive(VariableDeclaration const& _varDecl)
+{
+	solAssert(_varDecl.isConstant(), "Constant variable expected");
+	auto visitor = [&](VariableDeclaration const& _variable, util::CycleDetector<VariableDeclaration>& _cycleDetector, size_t _depth)
+	{
+		solAssert(_depth < 256, "Recursion depth limit reached");
+
+		if (auto identifier = dynamic_cast<Identifier const*>(_variable.value().get()))
+			if (auto referencedVarDecl = dynamic_cast<VariableDeclaration const*>(identifier->annotation().referencedDeclaration))
+				if (referencedVarDecl->isConstant())
+					if (_cycleDetector.run(*referencedVarDecl))
+						return;
+	};
+
+	if (util::CycleDetector<VariableDeclaration>(visitor).run(_varDecl))
+		return true;
+
+	return false;
```

Comment:
What about `return util::CycleDetector<VariableDeclaration>(visitor).run(_varDecl);` instead of 

```
	if (util::CycleDetector<VariableDeclaration>(visitor).run(_varDecl))
		return true;

	return false;
```

like in 
https://github.com/ethereum/solidity/blob/1df284733eb93e4b887c317b79412fe74fa7f731/libsolidity/analysis/PostTypeChecker.cpp#L204